src/processTargets/modifiers/scopeTypeStages/SubTokenStages.ts

```diff
@@ -0,0 +1,139 @@
+import { Range } from "vscode";
+import { SUBWORD_MATCHER } from "../../../core/constants";
+import { GRAPHEME_SPLIT_REGEX } from "../../../core/TokenGraphemeSplitter";
+import { Target } from "../../../typings/target.types";
+import {
+  ContainingScopeModifier,
+  EveryScopeModifier,
+} from "../../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../../typings/Types";
+import { matchAll } from "../../../util/regex";
+import { ModifierStage } from "../../PipelineStages.types";
+import { PlainTarget, SubTokenWordTarget } from "../../targets";
+import { getTokenRangeForSelection } from "./TokenStage";
+
+abstract class SubTokenStage implements ModifierStage {
+  constructor(
+    private modifier: ContainingScopeModifier | EveryScopeModifier,
+    private regex: RegExp
+  ) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    const tokenRange = getTokenRangeForSelection(
+      target.editor,
+      target.contentRange
+    );
+
+    const { document } = target.editor;
+    const text = document.getText(tokenRange);
+    const offset = document.offsetAt(tokenRange.start);
+
+    const contentRanges = matchAll<Range>(
+      text,
+      this.regex,
+      (match) =>
+        new Range(
+          document.positionAt(offset + match.index!),
+          document.positionAt(offset + match.index! + match[0].length)
+        )
+    );
+
+    const targets = this.createTargets(target, contentRanges);
+
+    if (this.modifier.type === "everyScope") {
+      return targets;
+    }
+
+    return [this.findSingleTarget(target, targets)];
+  }
+
+  private findSingleTarget(target: Target, targets: Target[]): Target {
+    let intersectingTargets = targets
+      .map((t) => ({
+        target: t,
+        intersection: t.contentRange.intersection(target.contentRange),
+      }))
+      .filter((it) => it.intersection != null);
+
+    // Empty range utilize single leftmost target
+    if (target.contentRange.isEmpty) {
+      return intersectingTargets[0].target;
+    }
+
+    // On non empty range utilize all non-empty intersecting targets
+    intersectingTargets = intersectingTargets.filter(
+      (it) => !it.intersection!.isEmpty
+    );
+
+    if (intersectingTargets.length === 1) {
+      return intersectingTargets[0].target;
+    }
+
+    return intersectingTargets[0].target.createContinuousRangeTarget(
+      target.isReversed,
+      intersectingTargets.at(-1)!.target,
+      true,
+      true
+    );
+  }
+
+  protected abstract createTargets(target: Target, ranges: Range[]): Target[];
+}
+
+export class WordStage extends SubTokenStage {
+  constructor(modifier: ContainingScopeModifier | EveryScopeModifier) {
+    super(modifier, SUBWORD_MATCHER);
+  }
+
+  protected createTargets(target: Target, ranges: Range[]): Target[] {
```

Comment:
I would call this `inputTarget`.  We're throwing the word "target" around a lot here so a bit of verbosity might aid clarity

I'd also add a docstring; not immediately obvious what this does and what `ranges` is supposed to be